packages/block-editor/src/hooks/block-style-variation.js

```diff
@@ -59,6 +59,85 @@ function getVariationNameFromClass( className, registeredStyles = [] ) {
 	return null;
 }
 
+/**
+ * This hook is used to generate new block style variation overrides
+ * based on an incoming theme config. If a matching style is found in the config,
+ * a new override is created and returned.
+ *
+ * @param {Object}               config             A global styles object, containing settings and styles.
+ * @param {Array<Array, Object>} variationOverrides An array of existing block variation overrides.
+ * @return {Array} An array of new block variation overrides.
+ */
+export function useUpdateBlockVariationOverridesWithConfig(
+	config,
+	variationOverrides = []
```

Comment:
There are probably opportunities to abstract this and share some love with `useBlockStyleVariation`. Perhaps return an override object instead. I'll look into it later - first on the agenda is to see if this PR flies at all.